fiber-bootstrap.ts

```diff
@@ -3,14 +3,20 @@
 
 import Fiber = require("fibers");
 import Future = require("fibers/future");
-import errors = require("./errors");
+// import errors = require("./errors");
 
-export function run(action: Function) {
-	Fiber(() => {
-		errors.installUncaughtExceptionListener();
+export function run(action: any) {
+	if(Fiber.current) {
+		// Use the already existing fiber, we do not need new one.
 		action();
-		$injector.dispose();
-		Future.assertNoFutureLeftBehind();
-	}).run();
+	} else {
+		Fiber(() => {
+			// TODO: Think about correct place for this.
+			// errors.installUncaughtExceptionListener();
```

Comment:
I think we can change errors.installUncaughtExceptionListener to be idempotent, this make it safe to be called many times. Internally, we can use _.once with the code of the existing function. After that,w e can uncomment the currently commented out code.